### ISSUE TITLE
We are trying to use sentiment text to make the generated audio have different emotions

### ISSUE THREAD

**Issue by xwan07017** [SWivid/F5-TTS#606]
### Checks

- [X] This template is only for feature request.
- [X] I have thoroughly reviewed the project documentation but couldn't find any relevant information that meets my needs.
- [X] I have searched for existing issues, including closed ones, and found not discussion yet.
- [X] I confirm that I am using English to submit this report in order to facilitate communication.

### 1. Is this request related to a challenge you're experiencing? Tell us your story.

We are trying to use sentiment text to make the generated audio have different emotions

### 2. What is your suggested solution?

We are trying to use sentiment text to make the generated audio have different emotions

### 3. Additional context or comments

We tried to add emotional features for training. The data set was an English data set of about 330 hours and trained for 200k steps, but the effect of emotional guidance was not very good. If you are interested, we can discuss it together.

### 4. Can you help us with this feature?

- [X] I am interested in contributing to this feature.

**Comment by SWivid** [comment]
Hi @xwan07017 , you mean to finetune with emotional data to enhance the ability and leverage the reference speech to control,
or with tokens that indicate each emotion to control

thought both will work, how is your observation (you may drop a zip in this issue or send with email if not proper here)
we have tried some other control with later method, and it works to some extent.

**Comment by xwan07017** [comment]
we input  reference speech and  emotion feature（like happy  or sad ...）， then we got  a  new audio with a happy emotion（or sad ...） and the same timbre as the reference speech.

**Comment by SWivid** [comment]
so you have introduce a new embedding concatenated (model structure modified) or the emotion features are served as new tokens
if the former one, it is like to train from scratch since input distribution changed (will need to train longer)
if later one, thought will work (but hard to diganose if no samples)

**Comment by xwan07017** [comment]
@SWivid  Yes, we  introduce a new embedding concatenated (model structure modified),  So we collected a new dataset and trained it from scratch.

**Comment by MithrilMan** [comment]
None of these solution would handle the case where I want to pass emotions between delimited text (e.g. xml style) like in the example below, correct?

e.g.
> <angry>How do you dare?</angry><sarcastic>Didn't your mother tell you not to play with fire?</sarcastic>

**Comment by niknah** [comment]
An alternative... 
https://github.com/niknah/ComfyUI-F5-TTS?tab=readme-ov-file#multi-voices
This is a custom node for ComfyUI.  

Speak in a sad voice, happy voice, record into different files.
Then use...
```
{happy} This is the narrator
{sad} Hello World this is the end
```

**Comment by xwan07017** [comment]
@SWivid Hey bro, Are you interested in joining us to discuss using emotions and text to make audio to have different emotions?

**Comment by SWivid** [comment]
> @SWivid Hey bro, Are you interested in joining us to discuss using emotions and text to make audio to have different emotions?

Sure, could contact me by email (listed in paper)

**Comment by xwan07017** [comment]
@SWivid  I have sent in email, please kindly check the email, thanks.

**Comment by amarbayar** [comment]
@SWivid Is SSML really possible in F5-TTS? :) I have been doing some reading and it seems like if the underlying model architecture doesn't support SSML natively then getting a support for SSML is pretty complex. 

Google's latest announcement covered their 2.0 model as well as "life-like TTS models" i.e Journey and they do support SSMLs pretty nice. If we can have something like that in F5-TTS that would be super. If there is anything I can do to contribute on this front, happy to spend some time.

**Comment by dafang** [comment]
Does F5-TTS currently support SSML? @SWivid

**Comment by francescocassini** [comment]
> Does F5-TTS currently support SSML? [@SWivid](https://github.com/SWivid)

Yes, I'm very interested in this feature...
Anyway I'm studying the possibility to generate SSML code from text and audio

**Comment by Ranjit246** [comment]
Any ideas or progress around this?
